### PR TITLE
NIFI-12271: Fix PutAzureBlobStorage - FileResourceService - FileNotFo…

### PR DESCRIPTION
**Pull request by Lehel44**
…und rollbacks

# Summary

[NIFI-12271](https://issues.apache.org/jira/browse/NIFI-12271)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java [hunk 1]
```diff
@@ -231,6 +231,7 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             getLogger().error("Failed to create blob on Azure Blob Storage", e);
             flowFile = session.penalize(flowFile);
             session.transfer(flowFile, REL_FAILURE);
+            session.getProvenanceReporter().send(flowFile, "/error/" + blobName);
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java:234]: The general purpose of provenance event reporting is to track successful operations as opposed to errors, so creating this event does not seem like the right approach. Is there a particular reason for this addition?

**Review by exceptionfactory — COMMENTED** [review]
Thanks for addressing this issue @Lehel44. Moving the `getFileResource()` method looks good, but I am concerned about the addition of the provenance event reporting as noted.

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java (hunk 1 above)
Comment (Lehel44) [reply nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java:234]: That was something I forgot to remove. Thanks!

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustment @Lehel44, the latest version looks good! +1 merging

**Pull request closed without merging**